src/rules/noTrailingWhitespaceRule.ts

```diff
@@ -78,12 +82,15 @@ function walk(ctx: Lint.WalkContext<Options>) {
     const sourceFile = ctx.sourceFile;
     const text = sourceFile.text;
     for (const line of getLineRanges(sourceFile)) {
-        const match = text.substr(line.pos, line.contentLength).match(/\s+$/);
-        if (match !== null) {
```

Comment:
You don't need a separate function or regex.
`match[0]` contains the full match. If the length of the match is equal to the line length, the line consists only of whitespace.
```ts
if (match !== null && (!ctx.options.skipBlankLines || match[0].length !== line.contentLength))
```

Reply:
or even simpler: check if `match.index === 0`